inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/StreamSinkServiceImpl.java

```diff
@@ -96,8 +98,14 @@ public Integer save(SinkRequest request, String operator) {
         // Make sure that there is no sink info with the current groupId and streamId
         String streamId = request.getInlongStreamId();
         String sinkType = request.getSinkType();
-        List<StreamSinkEntity> sinkExist = sinkMapper.selectByIdAndType(groupId, streamId, sinkType);
-        Preconditions.checkEmpty(sinkExist, ErrorCodeEnum.SINK_ALREADY_EXISTS.getMessage());
+        String sinkName = request.getSinkName();
+        List<StreamSinkEntity> sinkList = sinkMapper.selectByRelatedId(groupId, streamId);
```

Comment:
Maybe we can add sinkName into the select params, to reduce the number of the returned sink list.